src/crypto/tls/handshake_server_tls13.go

```diff
@@ -370,15 +410,45 @@ func (hs *serverHandshakeStateTLS13) pickCertificate() error {
 		}
 		return err
 	}
+
 	hs.sigAlg, err = selectSignatureScheme(c.vers, certificate, hs.clientHello.supportedSignatureAlgorithms)
 	if err != nil {
 		// getCertificate returned a certificate that is unsupported or
 		// incompatible with the client's signature algorithms.
 		c.sendAlert(alertHandshakeFailure)
 		return err
 	}
+
 	hs.cert = certificate
 
+	if hs.clientHello.delegatedCredentialSupported && len(hs.clientHello.supportedSignatureAlgorithmsDC) > 0 && c.config.GetDelegatedCredential != nil {
+		dCred, priv, err := c.config.GetDelegatedCredential(clientHelloInfo(c, hs.clientHello), nil)
+		if err != nil {
+			c.sendAlert(alertInternalError)
+			return nil
+		}
+
+		if dCred != nil && priv != nil {
+			hs.cert.PrivateKey = priv
+			if dCred.raw == nil {
+				dCred.raw, err = dCred.marshal()
+				if err != nil {
+					c.sendAlert(alertInternalError)
+					return err
+				}
+			}
+			hs.cert.DelegatedCredential = dCred.raw
+		}
+
+		hs.sigAlg, err = selectSignatureScheme(c.vers, certificate, hs.clientHello.supportedSignatureAlgorithmsDC)
```

Comment:
This should depend on the key of the delegated credential, not the certificate, unless I'm missing something?

Reply:
It is depending on the key of the delegated credential, as in few steps earlier the cert.PrivateKey is assigned the DC one:
```
			hs.cert.PrivateKey = priv
```